### PR TITLE
implement search and autocomplete

### LINKED ISSUE #11: Header - Autocomplete to search
Add autocomplete functionality to search bar in heaader. 

image

### PR DESCRIPTION
**Pull request by samuelopra**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #11

Test URLs:
- Before: https://main--hubblehomes-com--aemsites.hlx.live/
- After: https://issue-11--hubblehomes-com--aemsites.hlx.live/

### CODE REVIEW

**Comment by bhellema** [comment]
When a result is provided, and selecting the result I'm taken to a page of undefined.  Is this expected behaviour at this time?

**Comment by bhellema** [comment]
With all the formatting changes (which some should be reverted) has made finding the changes in this PR difficult.

File: scripts/dom-helpers.js [hunk 1]
```diff
@@ -64,44 +70,137 @@ export const removeEmptyTags = (block) => {
...
-export function nav(...items) { return domEl('nav', ...items); }
-export function aside(...items) { return domEl('aside', ...items); }
-export function meta(...items) { return domEl('meta', ...items); }
-export function picture(...items) { return domEl('picture', ...items); }
-export function br() { return domEl('br'); }
-export function select(...items) { return domEl('select', ...items); }
-export function option(...items) { return domEl('option', ...items); }
-export function dl(...items) { return domEl('dl', ...items); }
-export function dt(...items) { return domEl('dt', ...items); }
-export function dd(...items) { return domEl('dd', ...items); }
```
Comment (bhellema) [inline scripts/dom-helpers.js:107]: Please revert and put it back to this other format where it was much easier to read.

File: blocks/header/header.js [hunk 2]
```diff
@@ -78,62 +100,138 @@ async function buildNav() {
...
+
+    searchInput.addEventListener('input', handleInput);
+
+    document.addEventListener('click', (e) => {
+      if (e.target !== searchInput) {
+        autocompleteList.innerHTML = '';
+      }
+    });
+  } catch (error) {
+    console.error(error.message);
```
Comment (bhellema) [inline blocks/header/header.js:159]: console usage should cause failures in the build.

**Comment by samuelopra** [comment]
> When a result is provided, and selecting the result I'm taken to a undefined page. Is this expected behavior at this time?

this was happening for sales agents, but added a case for handling that now

File: blocks/header/header.js (hunk 2 above)
Comment (samuelopra) [reply blocks/header/header.js]: removed

File: scripts/dom-helpers.js (hunk 1 above)
Comment (samuelopra) [reply scripts/dom-helpers.js]: reverted to previous formatting

**Comment by samuelopra** [comment]
> With all the formatting changes (which some should be reverted) has made finding the changes in this PR difficult.

yep, sorry about that figured it was better than fixing a thousand lint errors each PR. next time, Ill be careful not to touch previously existing code.

**Comment by dave-fink** [comment]
Looks good with the changes.  I noticed the search icon got lost in the shuffle.  Can you check on this
image

**Comment by samuelopra** [comment]
> Looks good with the changes. I noticed the search icon got lost in the shuffle. Can you check on this image

It looks like this is happening on the main site already. For now, I increased the size and removed top padding

https://main--hubblehomes-com--aemsites.hlx.live/

**Review by varunmitra — APPROVED** [review]

**Pull request merged**